docs/playbook-reference/triggers/helm-releases-monitoring.rst

```diff
@@ -0,0 +1,158 @@
+    * ``duration``: This field specifies the threshold time, in seconds, before the trigger could be initiated, and the release status should continue to stay within this threshold time before the trigger is initiated. This field is optional.
+    * ``rate_limit``: Limit firing to once every `rate_limit` seconds. This field is optional.
+
+    .. admonition:: Example
+
+        This example demonstrates how to set up a trigger that watches a Helm release for an unhealthy state. If the release remains in a transient state for more than 1200 seconds, the `on_helm_release_unhealthy` trigger is activated.
+
+        .. code-block:: yaml
+
+            customPlaybooks:
+              - actions:
+                - helm_status_notification:
+                    message: "Helm release is in an unhealthy state" # optional
+                triggers:
+                  - on_helm_release_unhealthy:
+                      names: ["demo-app"] # optional
+                      namespace: "default" # optional
+                      duration: 1200 # optional
+                      rate_limit: 900 # optional
+
+        ``helm_status_notification`` supports the following parameters:
+            ``message``: Custom message for the action event. This field is optional.
+
+    .. image:: /images/helm-release-unhealthy.png
+      :width: 1000
+      :align: center
+
+
+.. _on_helm_release_fail:
+
+.. details:: on_helm_release_fail
+
+    The ``on_helm_release_fail`` event is triggered when a Helm release enters a ``failed`` state. This is a one-time trigger, meaning that it only fires once when the release fails.
+
+    **Available options**:
+
+    * ``names``: A list of Helm release names that this trigger will monitor. If you leave this field empty, the trigger will monitor all release names specified in the namespace. This field is optional.
+    * ``namespace``: This field specifies the namespace that the trigger will monitor. If you leave this field empty, the trigger will monitor across all namespaces. This field is optional.
+
+    .. admonition:: Example
+
+        This example demonstrates how to set up a trigger that watches a Helm release for an ``failed`` state.
+
+        .. code-block:: yaml
+
+            customPlaybooks:
+              - actions:
+                - helm_status_notification:
+                    message: "Helm release failed" # optional
+                triggers:
+                  - on_helm_release_fail:
+                      names: ["demo-app"] # optional
+                      namespace: "default" # optional
+
+        ``helm_status_notification`` supports the following parameters:
+        * ``message``: Custom message for the action event. This field is optional.
+
+    .. image:: /images/helm-release-failed.png
+      :width: 1000
+      :align: center
+
+.. _on_helm_release_deploy:
+
+.. details:: on_helm_release_deploy
```

Comment:
Does this mean that the Helm release was a success? Can deploy happen in case of failure or error?

Reply:
Please also apply comments from `on_helm_release_unhealthy` to this trigger.

Reply:
`on_helm_release_deploy` is only used in the case of successful deployment. Failed deployment will trigger `on_helm_release_fail`

Reply:
Thanks, can you clarify that in the description? It's really useful to know

Reply:
done.